### PR TITLE
test: adds unit tests for the toolbar title generator

### PR DESCRIPTION
**Pull request by jenifersantoszup**
Signed-off-by: jenifersantoszup <jenifer.santos@zup.com.br>

### Related Issues

#1192 

### Description and Example

Unit tests have been added to ensure that the generateTitle method is called correctly and that the defined title is correct in the toolbar.

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/fb9c8dffa193cd3692bda92024b56d5df6b0f71b/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L184">android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L184</a> - Prefer “<code>SHOW_AS_ACTION_IF_ROOM</code>” instead of “<code>SHOW_AS_ACTION_ALWAYS</code>”</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by carlossteinzup — DISMISSED** [review]

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt [hunk 1]
```diff
@@ -269,4 +237,34 @@ class ToolbarManagerTest : BaseTest() {
         verify(exactly = once()) { menuItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS) }
         verify(exactly = once()) { menuItem.icon = icon }
     }
+
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt:240]: please, use the new pattern for test, use the @DisplayName to given, when and then

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt [hunk 2]
```diff
@@ -269,4 +237,34 @@ class ToolbarManagerTest : BaseTest() {
...
+    @Test
+    fun `should be call the configureToolbar method to generate the title`() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt:242]: please, use the new pattern for test, use the @DisplayName to given, when and then

File: android/beagle/build.gradle [hunk 3]
```diff
@@ -90,13 +93,21 @@ dependencies {
...
-    testRuntimeOnly Dependencies.TestLibraries.junitEngine
-
     testImplementation Dependencies.TestLibraries.kotlinCoroutinesTest
     testImplementation Dependencies.TestLibraries.mockk
     testImplementation Dependencies.TestLibraries.archCoreTesting
     testImplementation Dependencies.GeneralLibraries.jsonObject
+    testImplementation Dependencies.TestLibraries.junitApi
+    testImplementation Dependencies.TestLibraries.junitEngine
+    testImplementation Dependencies.TestLibraries.robolectric
+    testImplementation Dependencies.TestLibraries.mockk
```
Comment (uziasferreirazup) [inline android/beagle/build.gradle:103]: duplicate dependency here, you can remove this line

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt [hunk 4]
```diff
@@ -36,73 +34,45 @@ import br.com.zup.beagle.android.components.layout.ScreenComponent
...
+import io.mockk.*
 import org.junit.jupiter.api.AfterEach
-import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertTrue
 import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
 
+@DisplayName("Given a toolbar")
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt:49]:
```suggestion
@DisplayName("Given a ToolbarManager")
```

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt [hunk 5]
```diff
@@ -269,4 +237,35 @@ class ToolbarManagerTest : BaseTest() {
         verify(exactly = once()) { menuItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS) }
         verify(exactly = once()) { menuItem.icon = icon }
     }
+
+    @Test
+    @DisplayName("should be call the configureToolbar method to generate the title")
+    fun `verifyCallTheGenerateTitle`() {
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt:243]:
```suggestion
    fun verifyCallTheGenerateTitle() {
```

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt [hunk 6]
```diff
@@ -36,73 +34,45 @@ import br.com.zup.beagle.android.components.layout.ScreenComponent
...
 import org.junit.jupiter.api.AfterEach
-import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertTrue
 import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
 
+@DisplayName("Given a toolbar")
 class ToolbarManagerTest : BaseTest() {
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarManagerTest.kt:50]: Please follow the guide line of tests:
https://github.com/ZupIT/beagle/wiki/Android-Tests-Style-Guide

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarTextManagerTest.kt [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Test
+import org.junit.jupiter.api.Assertions
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+
+@Config(manifest = Config.NONE)
+@RunWith(AndroidJUnit4::class)
+class ToolbarTextManagerTest {
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ToolbarTextManagerTest.kt:35]: Please follow the guideline of tests:
https://github.com/ZupIT/beagle/wiki/Android-Tests-Style-Guide

**Review by uziasferreirazup — CHANGES_REQUESTED** [review]

**Comment by uziasferreirazup** [comment]
I understand the robolectric test use junit4 but you can use the correct name to write tests `GIVEN ... WHEN... THEN` please read the documentation and adjust the code.

**Review by uziasferreirazup — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Pull request merged**